hail/src/main/scala/is/hail/expr/types/physical/PContainer.scala

```diff
@@ -134,14 +134,20 @@ abstract class PContainer extends PIterable {
   def setElementPresent(region: Code[Region], aoff: Code[Long], i: Code[Int]): Code[Unit] =
     setElementPresent(aoff, i)
 
+  def firstElementOffset(aoff: Long, length: Int): Long =
+    aoff + elementsOffset(length)
+
   def elementOffset(aoff: Long, length: Int, i: Int): Long =
-    aoff + elementsOffset(length) + i * elementByteSize
+    firstElementOffset(aoff, length) + i * elementByteSize
 
   def elementOffsetInRegion(region: Region, aoff: Long, i: Int): Long =
     elementOffset(aoff, loadLength(region, aoff), i)
 
   def elementOffset(aoff: Code[Long], length: Code[Int], i: Code[Int]): Code[Long] =
-    aoff + elementsOffset(length) + i.toL * const(elementByteSize)
+    firstElementOffset(aoff, length) + i.toL * const(elementByteSize)
+
+  def firstElementOffset(aoff: Code[Long], length: Code[Int]): Code[Long] =
```

Comment:
I like this :+1: I frequently end up doing something like `elementOffset(aoff, length, 0)` in NDArray code, having a function that just does that is nice. (Not related to this PR, but I've always found `elementOffset` and `elementsOffset` to be too similarly named for their very different return values. The name`elementsOffset` kind of sounds like it should do what `firstElementOffset` does, at least to me).

Reply:
I'm having a hard time with names here. I agree, we have some names that are offsets, and some are addresses (those that include the address/offset of that element in the region).

I propose *Addrss for the latter, e.g firstElementAddress